cg/services/pacbio/metrics/models.py

```diff
@@ -101,3 +107,31 @@ class PolymeraseMetrics(BaseModel):
     _validate_longest_subread_length_n50 = field_validator(
         "longest_subread_length_n50", mode="before"
     )(divide_by_thousand_with_one_decimal)
+
+
+class SmrtlinkDatasetsMetrics(BaseModel):
+    """Model to parse metrics in the SMRTlink datasets report."""
+
+    device_internal_id: str = Field(..., alias=SmrtLinkDatabasesIDs.CELL_ID)
+    well_name: str = Field(..., alias=SmrtLinkDatabasesIDs.WELL_NAME)
+    well_sample_name: str = Field(..., alias=SmrtLinkDatabasesIDs.WELL_SAMPLE_NAME)
+    sample_internal_id: str = Field(..., alias=SmrtLinkDatabasesIDs.BIO_SAMPLE_NAME)
+    movie_name: str = Field(..., alias=SmrtLinkDatabasesIDs.MOVIE_NAME)
+    cell_index: int = Field(..., alias=SmrtLinkDatabasesIDs.CELL_INDEX)
+    path: str = Field(..., alias=SmrtLinkDatabasesIDs.PATH)
+
+    @property
+    def plate(self) -> int:
+        """
+        Extract the plate number from the plate_well name in the path.
+        Example:
+            /1_A01/ -> plate = 1
+            /2_D01/ -> plate = 2
+        """
+        match = re.search(r"/(\d)_", self.path)
+        if not match:
+            raise ValidationError("Path should contain the plate-well name")
+        plate = int(match.group(1))
```

Comment:
This seems quite sketchy. If we search the whole path I think we should have a more exact regexp

Reply:
I also though about this:
```python
@property
    def plate(self) -> int:
        """
        Extract the plate number from the plate_well name in the path.
        Example:
            /1_A01/ -> plate = 1
            /2_D01/ -> plate = 2
        """
        path_parts: list[str] = self.path.split("/")
        plate_well: str = [part for part in path_parts if self.well_name in part][0]
        plate = int(plate_well[0])
        if plate not in [1, 2]:
            raise ValidationError("Value of 'plate' should be 1 or 2")
        return plate
```

Reply:
updated to model validator according to documentation